spec/unit/deployment.spec.js

```diff
@@ -0,0 +1,282 @@
+
+        var tool = deployment.getDeploymentTool('10.0');
+        expect(tool instanceof WinAppDeployCmdTool).toBe(true);
+
+    });
+
+});
+
+describe('Windows 10 deployment interacts with the file system as expected.', function() {
+
+    function runMock(cmd, args, cwd) {
+        expect(cmd).toBe('c:\\Program Files (x86)\\Windows Kits\\10\\bin\\x86\\WinAppDeployCmd.exe');
+        switch (args[0]) {
+            case 'devices':
+                var output = 'Windows App Deployment Tool\r\nVersion 10.0.0.0\r\nCopyright (c) Microsoft Corporation. All rights reserved.\r\n\r\nDiscovering devices...\r\nIP Address      GUID                                    Model/Name\r\n127.0.0.1   00000015-b21e-0da9-0000-000000000000    Lumia 1520 (RM-940)\r\n10.120.70.172   00000000-0000-0000-0000-00155d619532    00155D619532\r\n10.120.68.150   00000000-0000-0000-0000-00155d011765    00155D011765\r\nDone.';
+                return Q(output);
+
+            case 'update':
+            case 'install':
+                expect(args[2]).toBe(TEST_APP_PACKAGE_NAME);
+                expect(args[4]).toBe('127.0.0.1');
+                return Q('');
+
+            case 'uninstall':
+                expect(args[2]).toBe(TEST_APP_PACKAGE_ID);
+                expect(args[4]).toBe('10.120.68.150');
+                return Q('');
+
+        }
+    }
+
+    var mockedProgramFiles = process.env['ProgramFiles(x86)'];
+    beforeEach(function() {
+        deployment.__set__('run', runMock);    
+        process.env['ProgramFiles(x86)'] = 'c:\\Program Files (x86)';
+    });
+    afterEach(function() {
+        deployment.__set__('run', run);
+        if (mockedProgramFiles) {
+            process.env['ProgramFiles(x86)'] = mockedProgramFiles;
+        } else {
+            delete process.env['ProgramFiles(x86)'];
+        }
+    });
+
+    it('enumerateDevices returns a valid set of objects', function() {
+        var deploymentTool = deployment.getDeploymentTool('10.0');
+        var done = false;
+        deploymentTool.enumerateDevices().then(function(deviceList) {
+
+            expect(deviceList.length).toBe(3);
+            expect(deviceList[0].name).toBe('Lumia 1520 (RM-940)');
+            expect(deviceList[0].index).toBe(0);
+            expect(deviceList[0].type).toBe('device');
+            
+            done = true;
+
+        });
+
+        waitsFor(function() { return done; });
+    });
+
+    it('installAppPackage passes the correct set of parameters', function() {
+        var deploymentTool = deployment.getDeploymentTool('10.0');
+        var done = false;
+        deploymentTool.enumerateDevices().then(function(deviceList) {
+            deploymentTool.installAppPackage(TEST_APP_PACKAGE_NAME, deviceList[0], false, false).then(function() {
+
+                // expect() calls are in the runMock function
+                done = true;
+
+            });
+        });
+
+        waitsFor(function() { return done; });
```

Comment:
Is this the old way of doing async on Jasmine?. We should really upgrade to newer version of Jasmine as it offers better async support.